packages/react-tooltip/Spec.md

```diff
@@ -1,63 +1,474 @@
+<Button tooltip={{ children: 'Placed Tooltip', placement: 'right' }} />
+<Button tooltip={{ as: MyTooltip, children: 'You can even implement your own tooltip' }} />
+```
+
+## On any element using a ref
+
+To attach a tooltip to a component that doesn't have a `tooltip` slot, use the `useTooltipRef` hook. It takes the same shorthand props that the `tooltip` slot does. This is slightly less ergonomic than the tooltip slot, and requires using a ref, but works with any element.
+
+```jsx
+<div ref={useTooltipRef('Example tooltip', { id: 'exampleTooltipId' })} aria-describedby="exampleTooltipId">
+  A div with a tooltip
+</div>
+
+<ThirdPartyComponent ref={useTooltipRef(<>Tooltips <u>everywhere</u>!</>)} />
+```
+
+# Variants
+
+- The tooltip can have a `subtle` style variant with a different background and text color.
+- The tooltip can optionally render an arrow pointing to the target element (on by default).
+
+# API
+
+The Tooltip API is split among several components and hooks, in two packages. Having two packages allows the lightweight `react-tooltip-provider` package to be referenced by any component, while the bulk of the code for tooltips lives in the larger `react-tooltip` package.
+
+- [**@fluentui/react-tooltip-provider**](#fluentuireact-tooltip-provider)
+  - [`TooltipProps`](#tooltipprops)
+  - [`TooltipManagerApi`](#tooltipmanagerapi)
+  - [`TooltipProvider`](#tooltipprovider)
+  - [`useTooltipSlot`](#usetooltipslot)
+  - [`useTooltipRef`](#usetooltipref)
+- [**@fluentui/react-tooltip**](#fluentuireact-tooltip)
+  - [`Tooltip`](#tooltip)
+  - [`TooltipManager`](#tooltipmanager)
+
+_A note about the terminology used for the elements that the tooltip is attached to:_
+
+- _The **trigger** is the element that causes the tooltip to open._
+- _The **target** is the element that the tooltip is anchored to (and the arrow points to)._
+
+_Almost always, these will both be the same element, but it is possible to specify them separately, so the tooltip can show up adjacent to a different element than the one that triggered it._
+
+## @fluentui/react-tooltip-provider
+
+The `react-tooltip-provider` is a lightweight package that allows any component to hook into tooltip functionality.
+
+### TooltipProps
+
+The `TooltipProps` interface is defined in `react-tooltip-provider` so that components can specify the details of the tooltip without taking a dependency on the larger `react-tooltip` package.
+
+```typescript
+export interface TooltipProps extends ComponentProps, React.HTMLAttributes<HTMLElement> {
+  /**
+   * How to position the tooltip relative to the target element. This is a "best effort" placement,
+   * but the tooltip may be flipped to the other side if there is not enough room.
+   *
+   * @defaultvalue bottom
+   */
+  placement?: TooltipPlacement;
+
+  /**
+   * Color variant with a subtle look
+   */
+  subtle?: boolean;
+
+  /**
+   * Do not render an arrow pointing to the target element
+   */
+  noArrow?: boolean;
```

Comment:
v0 uses `pointing` for this (opposite of `noArrow`/`arrowless`). Which do people like the most between them?

Reply:
Oh yeah, I could totally see that, I guess it depends on what we want the default to be since I think we want the default value of these props to be false.

Reply:
are we going to allow custom styling on the arrow ? potentially would qualify that as a slot ?